lighthouse-core/gather/driver.js

```diff
@@ -490,14 +488,53 @@ class Driver {
   }
 
   /**
-   * Navigate to the given URL. Use of this method directly isn't advised: if
+   * Set up listener for network quiet events and URL redirects. URL in options
+   * object will be updated to the post-redirected URL.
+   * @param {{url: string}} options
+   * @return {!Promise}
+   * @private
+   */
+  _beginNetworkMonitoring(options) {
+    return new Promise((resolve, reject) => {
+      this._networkStatus = new NetworkRecorder([], this);
+      this.enableUrlUpdateIfRedirected(options);
+
+      this.sendCommand('Network.enable').then(resolve, reject);
+    });
+  }
+
+  /**
+   * Dispatch relevant events to the networkStatus listener.
+   * @param {string} method
+   * @param {!Object<string, *>=} params
+   * @private
+   */
+  _recordNetworkEvent(method, params) {
+    if (!this._networkStatus) return;
+
+    const regexFilter = /^Network\./;
+    if (!regexFilter.test(method)) return;
```

Comment:
I just moved it into `NetworkRecorder.dispatch` (to mirror `devtools-log.record`), though we could arguably drop it altogether because the switch statement rejects anything not recognized